extensions/oplogPopulator/KafkaConnectWrapper.js

```diff
@@ -0,0 +1,307 @@
+const { errors } = require('arsenal');
+const joi = require('joi');
+const http = require('http');
+
+const paramsJoi = joi.object({
+    kafkaConnectHost: joi.string().required(),
+    kafkaConnectPort: joi.number().required(),
+    logger: joi.object().required(),
+}).required();
+
+/**
+ * @class KafkaConnectWrapper
+ *
+ * @classdesc Wrapper arround the Kafka Connect REST API.
+ */
+class KafkaConnectWrapper {
+
+    /**
+     * @constructor
+     * @param {Object} params constructor params
+     * @param {string} params.kafkaConnectHost kafka connect host
+     * @param {number} params.kafkaConnectPort kafka connect port
+     * @param {Logger} params.logger - logger object
+     */
+    constructor(params) {
+        joi.attempt(params, paramsJoi);
+        this._kafkaConnectHost = params.kafkaConnectHost;
+        this._kafkaConnectPort = params.kafkaConnectPort;
+        this._logger = params.logger;
+    }
+
+    /**
+     * makes http request
+     * @param {Object} params request params
+     * @param {string} params.method http method
+     * @param {string} params.path path to resource
+     * @param {string} params.headers request headers (optional)
+     * @param {Object} data request data
+     * @returns {Promise|Object} http response
+     */
+    makeRequest(params, data) {
```

Comment:
in preparation for the coming typescript migration, we may want to start "typing" (or pre-typing in comments) these kind of parameters....

(or maybe we should have added this file as typescript already, btw? maybe we'll try that next time...)